### DISCUSSION TITLE
Running queries via weblite sql

### DISCUSSION THREAD

**Discussion by wozdog-oz** [sqlitecloud/discussions#29] [General]
The weblite REST api seems to work for most of the api (e.g. querying rows from a table) both using postman and via the dashboard weblite documentation,  however I can't get the /v2/weblite/sql  endpoint to work. 

I get the following error with any SQL in the body:
        "title": "POST /v2/weblite/sql Bad request"

Any suggestions as to what may be going wrong?

thanks

**Comment by TizianoT** [comment]
Hi, thanks for your question.

The `/v2/weblite/sql` endpoint can be accessed using an authenticated GET or POST request.

To authenticate the request, you can use the project's API key, which can be passed in one of the following ways:

1. As a query string: `/v2/weblite/sql?apikey=myapikey`
2. As a bearer authorization header: `Authorization: Bearer myapikey`

Then, you need to provide the SQL statement to be executed and specify the database on which to execute the SQL statement.

Using a **GET request**:
`/v2/weblite/sql?apikey=myapikey&database=chinook.sqlite&sql=SELECT * FROM artists LIMIT 10`

Using a **POST request**:
`/v2/weblite/sql?apikey=myapikey`

With the body:
```
{
   "sql": "SELECT * FROM artists LIMIT 10",
   "database": "chinook.sqlite"
}

```

**Comment by wozdog-oz** [comment]
Ah that is what was missing – when doing a POST, the body needed to be JSON not just the plain-text sql.

Thanks!


From: TizianoT ***@***.***>
Date: Friday, 19 July 2024 at 20:43
To: sqlitecloud/discussions ***@***.***>
Cc: wozdog ***@***.***>, Author ***@***.***>
Subject: Re: [sqlitecloud/discussions] Running queries via weblite sql (Discussion #29)

Hi, thanks for your question.

The /v2/weblite/sql endpoint can be accessed using an authenticated GET or POST request.

To authenticate the request, you can use the project's API key, which can be passed in one of the following ways:

  1.  As a query string: /v2/weblite/sql?apikey=myapikey
  2.  As a bearer authorization header: Authorization: Bearer myapikey

Then, you need to provide the SQL statement to be executed and specify the database on which to execute the SQL statement.

Using a GET request:
/v2/weblite/sql?apikey=myapikey&database=chinook.sqlite&sql=SELECT * FROM artists LIMIT 10

Using a POST request:
/v2/weblite/sql?apikey=myapikey

With the body:

{

   "sql": "SELECT * FROM artists LIMIT 10",

   "database": "chinook.sqlite"

}

**Comment by mayneyao** [comment]
@TizianoT  can we add support for `bind` params? 

```json
{
   "sql": "SELECT * FROM artists where name = ?  LIMIT 10",
   "bind": ["alice"]
   "database": "chinook.sqlite"
}
```

**Reply by TizianoT** [reply]
We don't currently support bind parameters, but we plan to implement that soon and will let you know when it's ready.
If you need to execute an SQL query with bind parameters, you can use our edge function features, which make Query parameters and values in your request body available with the '$' prefix.

```
SELECT * FROM artists
WHERE name LIKE $filter || '%'
LIMIT $limit
```

<img width="1474" alt="image" src="https://github.com/user-attachments/assets/0e00fae8-f8a6-4d0d-a50f-b89dda754760">
